datasets/datasets_openml.py

```diff
@@ -46,8 +49,16 @@ def get_data(self):
             self.dataset
         )
         X, y, cat_indicator, attribute_names = dataset.get_data(
-            dataset_format="array", target=dataset.default_target_attribute
+            dataset_format="dataframe", target=dataset.default_target_attribute
         )
+        # convert int and object to categories
+        # TODO: handle cases where categorical cardinality is > 255
+        # (this is not supported using HGBT native categorical handling)
+        for col in X.columns:
+            is_integer = pd.api.types.is_integer_dtype(X[col])
```

Comment:
This should be with `and X[col].nunique() < 255`?

Reply:
(I removed the comment). I think it's safer and easier to also convert high-cardinality to categories for a first version, and let the model fail if it doesn't support it (like HGBT). For these models, you would need specific preprocessing for these features. (But these high-cardinality features are not present in my classification benchmark.)

Reply:
Ok, but to me, the danger is that you convert an integer numerical value (let's say age) into a category?